File: frontend/src/services/cards.ts
import { apiService } from "@/lib/api";

interface StatusAvaliacoes {
  faltam_1_avaliacao: number;
  faltam_2_avaliacoes: number;
  faltam_3_avaliacoes: number;
}

interface CardsData {
  total_projetos: number;
  trabalhos_para_avaliar: number;
  trabalhos_avaliados: number;
  avaliadores_ativos: number;
  progresso_geral: number;
  status_avaliacoes: StatusAvaliacoes;
}

class CardsService {
  async getCardsData(): Promise<CardsData> {
    try {
      const response = await apiService.get<CardsData>('/cards');

      if (response) {
        return response;
      } else {
        throw new Error('Erro ao buscar dados dos cards');
      }
    } catch (error) {
      console.error('Erro ao buscar dados dos cards:', error);
      throw new Error('Erro ao buscar dados dos cards');
    }
  }
}

export const cardsService = new CardsService();
export type { CardsData, StatusAvaliacoes }; 

File: backend/app/schemas/cards.py
from pydantic import BaseModel
from typing import Dict

class StatusAvaliacoes(BaseModel):
    faltam_1_avaliacao: int
    faltam_2_avaliacoes: int
    faltam_3_avaliacoes: int

class CardsResponse(BaseModel):
    total_projetos: int
    trabalhos_para_avaliar: int
    trabalhos_avaliados: int
    avaliadores_ativos: int
    progresso_geral: int
    progresso_geral_inicial: int
    status_avaliacoes: StatusAvaliacoes 